### PR TITLE
split DEBUG and ERROR logs

### PR DESCRIPTION
**Pull request by X4mp**
debug logs are now logged to /opt/tanner/tanner.log
error logs are now logged to /opt/tanner/tanner.err

### CODE REVIEW

File: tanner/tests/test_config.py [hunk 1]
```diff
@@ -13,7 +13,8 @@ def setUp(self):
 				  'REDIS': {'host': 'localhost', 'port': '1337', 'poolsize': '40', 'timeout': '5'},
 				  'EMULATORS': {'root_dir': '/tmp/user_tanner'},
 				  'SQLI': {'db_name': 'user_tanner.db'},
-				  'LOGGER': {'log_file': '/tmp/user_tanner/user_tanner.log'},
+				  'DEBUG_LOGGER': {'log_file': '/tmp/user_tanner/user_tanner.log'},
+				  'ERROR_LOGGER': {'log_file': '/tmp/user_tanner/user_tanner.err'},
```
Comment (rnehra01) [inline tanner/tests/test_config.py]: Logs are stored in `/opt/tanner` directory.

File: tanner/utils/logger.py [hunk 2]
```diff
@@ -4,16 +4,22 @@
...
                                       datefmt='%Y-%m-%d %H:%M')
-        handler.setFormatter(formatter)
 
-        logger.addHandler(handler)
+        # DEBUG log to 'tanner.log'
+        debug_log_handler = logging.handlers.RotatingFileHandler(debug_filename, encoding='utf-8')
+        debug_log_handler.setFormatter(formatter)
+
+        # ERROR log to 'tanner.err'
+        error_log_handler = logging.FileHandler(err_filename, encoding='utf-8')
```
Comment (rnehra01) [inline tanner/utils/logger.py]: I think we can use `RotatingFileHandler` for this too. What do you think?

**Comment by rnehra01** [comment]
Just looked at this, actually I talked to @afeena about log files yesterday, when found the solution, just made the PR(https://github.com/mushorg/tanner/pull/127). I'll close my PR once you fix the tests.

**Comment by X4mp** [comment]
@rnehra01 thanks for the feedback. I'll fix the tests asap.

File: tanner/config.py [hunk 3]
```diff
@@ -11,7 +11,8 @@
                    'REDIS': {'host': 'localhost', 'port': 6379, 'poolsize': 80, 'timeout': 1},
                    'EMULATORS': {'root_dir': '/opt/tanner'},
                    'SQLI': {'db_name': 'tanner.db'},
-                   'LOGGER': {'log_file': '/opt/tanner/tanner.log'},
+                   'DEBUG_LOGGER': {'log_file': '/opt/tanner/tanner.log'},
+                   'ERROR_LOGGER': {'log_file': '/opt/tanner/tanner.err'},
```
Comment (rnehra01) [inline tanner/config.py]: I think making LOGGER as parent would make the code look better. Something like this
`'LOGGER': {'error_log_file': '/opt/tanner/tanner_errors.log', 'event_log_file': '/opt/tanner/tanner_events.log'}` in place of line 14 and 15. What do you say?

Comment (X4mp) [inline tanner/config.py]: like in your PR. I didn't want to copy so I left it my way. But I'll fix it now ;)

Comment (rnehra01) [inline tanner/config.py]: Well, It won't be treated as a copy. Just for making the code better, we can suggest some things to each other. :)

**Comment by rnehra01** [comment]
@X4mp It is a nice practice to submit a PR for an issue in form of single commit so that all the relevant changes will be stored in a single commit and can be easy to track.

**Comment by rnehra01** [comment]
@afeena I think we should document this also, how to make a contribution to this project. What's say?
Something like this https://github.com/zaproxy/zaproxy/blob/develop/CONTRIBUTING.md#what-we-zap-team-expect-from-you

**Comment by rnehra01** [comment]
@X4mp Wait for @afeena to take a look.

File: tanner/utils/logger.py [hunk 4]
```diff
@@ -4,16 +4,22 @@
...
-        handler = logging.handlers.RotatingFileHandler(log_filename, encoding='utf-8')
-
         formatter = logging.Formatter(fmt='%(asctime)s %(levelname)s:%(name)s:%(funcName)s: %(message)s',
                                       datefmt='%Y-%m-%d %H:%M')
-        handler.setFormatter(formatter)
 
-        logger.addHandler(handler)
+        # DEBUG log to 'tanner.log'
+        debug_log_handler = logging.handlers.RotatingFileHandler(debug_filename, encoding='utf-8')
+        debug_log_handler.setFormatter(formatter)
```
Comment (afeena) [inline tanner/utils/logger.py]: Maybe we need to set level for debug_log_handler? Errors are still in the tanner.log :)
```
afeena@ubuntu:/opt/tanner$ tail -n 2 tanner.log
2017-03-23 10:58 ERROR:tanner.rfi_emulator.RfiEmulator:get_rfi_result: Error during connection to php sandbox [Errno 111] Cannot connect to host 127.0.0.1:8088 ssl:False [Can not connect to 127.0.0.1:8088 [Connection refused]]
2017-03-23 10:58 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"version": 1, "response": {"message": {"sess_uuid": "6fb6697d-101f-44df-8cb9-605d7ebdde91", "detection": {"payload": "", "order": 2, "name": "rfi"}}}}'

afeena@ubuntu:/opt/tanner$ tail -n 1 tanner.err
2017-03-23 10:58 ERROR:tanner.rfi_emulator.RfiEmulator:get_rfi_result: Error during connection to php sandbox [Errno 111] Cannot connect to host 127.0.0.1:8088 ssl:False [Can not connect to 127.0.0.1:8088 [Connection refused]]
```

File: tanner/config.py [hunk 5]
```diff
@@ -11,7 +11,7 @@
                    'REDIS': {'host': 'localhost', 'port': 6379, 'poolsize': 80, 'timeout': 1},
                    'EMULATORS': {'root_dir': '/opt/tanner'},
                    'SQLI': {'db_name': 'tanner.db'},
-                   'LOGGER': {'log_file': '/opt/tanner/tanner.log'},
+                   'LOGGER': {'log_debug': '/opt/tanner/tanner.log', 'log_err': '/opt/tanner/tanner.err'},
```
Comment (afeena) [inline tanner/config.py]: How do you think, should debug logs be optional? Or we should provide it by default? Question for discussion :)

Comment (rnehra01) [inline tanner/config.py]: I think we can make debug logs optional and use something called `event logs` for events.

File: tanner/utils/logger.py (hunk 4 above)
Comment (X4mp) [inline tanner/utils/logger.py]: Ah ok, it's different from what I assumed. This is caused by the log level hirachy. I'll inspect how to handle it properly.

File: tanner/config.py (hunk 5 above)
Comment (X4mp) [inline tanner/config.py]: I would leave debug logs as they are and provide it by default since most of the logs go to the debug logfile anyway.

Comment (Philelis) [inline tanner/config.py]: I think for debugging it is good to have them, so it would be nice to turn it on and off.

**Comment by glaslos** [comment]
What is the status of this PR?

**Comment by X4mp** [comment]
AFAIK is @afeena testing 2 variants of this PR to find the better one.

File: tanner/config.py [hunk 6]
```diff
@@ -11,9 +11,10 @@
...
                    'MONGO': {'enabled': 'False', 'URI': 'mongodb://localhost'},
-                   'LOCALLOG': {'enabled': 'False', 'PATH': '/tmp/tanner_report.json'}
+                   'LOCALLOG': {'enabled': 'False', 'PATH': '/tmp/tanner_report.json'},
+                   'CLEANLOG': {'enabled': 'True'}
```
Comment (afeena) [inline tanner/config.py]: Maybe we should make it `false` by default?

**Comment by afeena** [comment]
We discussed storing the context of the error. 

Sure, we have different errors. Some of them are obvious, but other should provide additional info. e.g.
```
2017-04-26 00:27 ERROR:tanner.rfi_emulator.RfiEmulator:get_rfi_result: Error during connection to php sandbox [Errno 111] Cannot connect to host 127.0.0.1:8088 ssl:False [Can not connect to 127.0.0.1:8088 [Connection refused]]
2017-04-26 00:28 ERROR:tanner.rfi_emulator.RfiEmulator:download_file: Error during downloading the rfi script [Errno -5] Cannot connect to host foo:80 ssl:False [No address associated with hostname]
```

But we can add additional info to store the context
```
2017-04-26 00:30 ERROR:tanner.rfi_emulator.RfiEmulator:download_file: Error during downloading the rfi script by url http://foo/  [Errno -5] Cannot connect to host foo:80 ssl:False [No address associated with hostname]
```
In the second example, we don't need full info about paths. We just add additional data and the error becomes obvious.

I still think that splitting logs is a good idea. Because often debug info is off in production. And we should be aware only about errors.

Also if we have debug logs mixed up with errors, we can missed the error. e.g. 
```
2017-03-14 05:37 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 05:51 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 05:51 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 06:00 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /
2017-03-14 06:00 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 06:43 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /
2017-03-14 06:43 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 06:48 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 06:48 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 06:49 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /
2017-03-14 06:49 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 07:13 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /
2017-03-14 07:13 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 07:17 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 07:17 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 07:20 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 07:20 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 07:46 ERROR:tanner.session_analyzer.SessionAnalyzer:analyze: Can't get session for analyze: Expecting value: line 1 column 1 (char 0)
2017-03-14 07:46 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 07:46 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 07:49 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 07:49 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 08:00 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 08:00 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 08:00 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 08:00 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 08:18 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /
2017-03-14 08:18 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 08:22 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 08:22 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 08:27 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /
2017-03-14 08:27 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 08:29 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path /wp-login.php
2017-03-14 08:29 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 1, "name": "index"}}}, "version": 1}'
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path http://163.172.157.231:80/mysql/admin/
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 0, "name": "unknown"}}}, "version": 1}'
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path http://163.172.157.231:80/mysql/sqlmanager/
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 0, "name": "unknown"}}}, "version": 1}'
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path http://163.172.157.231:80/mysql/mysqlmanager/
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 0, "name": "unknown"}}}, "version": 1}'
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path http://163.172.157.231:80/phpmyadmin/
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 0, "name": "unknown"}}}, "version": 1}'
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path http://163.172.157.231:80/phpMyadmin/
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 0, "name": "unknown"}}}, "version": 1}'
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: Requested path http://163.172.157.231:80/phpMyAdmin/
2017-03-14 08:41 INFO:tanner.server.HttpRequestHandler:handle_event: TANNER response b'{"response": {"message": {"detection": {"order": 0, "name": "unknown"}}}, "version": 1}'
```

And file contains much more  lines ;) I vote for splitting! ;)

**Pull request merged**